packages/web/src/components/twitter-auth/TwitterAuth.tsx

```diff
@@ -0,0 +1,173 @@
+import { CSSProperties, MouseEventHandler, ReactNode } from 'react'
+
+import 'whatwg-fetch'
+import 'url-search-params-polyfill'
+
+export type TwitterAuthProps = {
+  children?: ReactNode
+  className?: string
+  credentials?: 'omit' | 'same-origin' | 'include'
+  customHeaders?: Record<string, any>
+  dialogHeight?: number
+  dialogWidth?: number
+  forceLogin?: boolean
+  loginUrl: string
+  onClick: () => void
+  onFailure: (error: any) => void
+  onSuccess: (response: Body) => void
+  requestTokenUrl: string
+  screenName?: string
+  style?: CSSProperties
+  text?: string
+}
+
+const messages = {
+  signIn: 'Sign in with Twitter'
+}
+
+const TwitterAuth = (props: TwitterAuthProps) => {
+  const {
+    children,
+    className,
+    credentials = 'same-origin',
+    customHeaders,
+    dialogHeight = 400,
+    dialogWidth = 600,
+    forceLogin = false,
+    loginUrl,
+    onClick,
+    onFailure,
+    onSuccess,
+    requestTokenUrl,
+    screenName,
+    style,
+    text = messages.signIn
+  } = props
+
+  const onButtonClick: MouseEventHandler<HTMLDivElement> = (e) => {
+    e.preventDefault()
+    onClick?.()
+    return getRequestToken()
+  }
+
+  const getHeaders = () => {
+    return { ...customHeaders, 'Content-Type': 'application/json' }
+  }
+
+  const getRequestToken = () => {
```

Comment:
Async here or are we cool with .then

Reply:
I tried to limit to just converting js -> ts, wanted to avoid more risky changes